Backend/controllers/events.js

```diff
@@ -25,20 +26,83 @@ dotenv.config();
  * @throws {Error} - A call to next() on an internal server error
  * @description This function creates an event and saves it to the database.
  */
-export const createEvent = async (req, res, next) => {
+
+// export const createEvent = async (req, res, next) => {
+//   try {
+//     const eventData = req.body; // Assuming the event data is sent in the request body
+//     const event = await Event.create(eventData);
+//     if (!event) {
+//       const error = new Error(
+//         "An error occurred while creating the event. Please try again."
+//       );
+//       error.statusCode = 400;
+//       throw error;
+//     }
+//     return res.status(201).json(event);
+//   } catch (error) {
+//     const Event = require("../models/event");
+
+//     return next(error);
+//   }
+// };
+
+// Create a new event
+exports.createEvent = async (req, res) => {
   try {
-    const eventData = req.body; // Assuming the event data is sent in the request body
-    const event = await Event.create(eventData);
-    if (!event) {
-      const error = new Error(
-        "An error occurred while creating the event. Please try again."
-      );
-      error.statusCode = 400;
-      throw error;
-    }
-    return res.status(201).json(event);
+    const { title, description, date, organizer } = req.body;
+
+    const event = new Event({
+      title,
+      description,
+      date,
+      organizer,
+      attendance: 0,
+      ticketSales: 0,
+      revenue: 0,
+      ratings: 0,
+    });
+
+    await event.save();
+
+    res.status(201).json(event);
   } catch (error) {
-    return next(error);
+    res.status(500).json({ error: "Failed to create event" });
+  }
+};
+
+// Get events for a specific organizer
+exports.getOrganizerEvents = async (req, res) => {
+  try {
+    const { organizerId, startDate, endDate } = req.query;
+
+    const query = {
+      organizer: organizerId,
+      date: { $gte: new Date(startDate), $lte: new Date(endDate) },
+    };
+
+    const events = await Event.find(query);
+
+    res.json(events);
+  } catch (error) {
+    res.status(500).json({ error: "Failed to get organizer events" });
+  }
+};
+
+// Get event analytics
+exports.getEventAnalytics = async (req, res) => {
```

Comment:
I was thinking we can approach calculating the number of tickets sold and the total revenue dynamically without storing these fields in the database for the corresponding ticket data since we already have a tickets field in our event schema. To calculate these, we may just get the length of the tickets field to obtain the number of participants for the event, then multiply that with the event price to get the total revenue generated by the event.